server/conn.go

```diff
@@ -285,6 +286,7 @@ func parseAttrs(data []byte) (map[string]string, error) {
 }
 
 func (cc *clientConn) readHandshakeResponse() error {
+	remoteAddr := cc.server.remoteAddr(cc.conn)
```

Comment:
According to spec, we need to abort the connection if there are errors when parsing the PROXY protocol header.

> Any sequence which does not exactly match the protocol must be discarded and
cause the receiver to abort the connection. It is recommended to abort the
connection as soon as possible so that the sender gets a chance to notice the
anomaly and log it.